### DISCUSSION TITLE
How to keep/save bookmarks during `git rebase` when file content doesn't change?

### DISCUSSION THREAD

**Discussion by Tony2** [alefragnani/vscode-bookmarks#429] [Q&A] [answered]
I have some bookmarks for a file in a git project. When I update the project using `git fetch; git rebase`,  the bookmarks are gone and I must add them again, manually. 

I am aware that the extension doesn't work well with external changes to the file (e.g. Prettier). In my case the file content doesn't change after the git update, so I hope it's possible to keep the bookmarks?

**Comment by alefragnani** [comment, marked as answer]
Hi @Tony2 ,

If the file doesn’t change, the bookmarks shouldn’t change as well. If you open the file _after the `git fetch`_, it will validate if the bookmarks are still valid (line numbers), but just that. 

Unless, when you run `git rebase`, the bookmarked file is opened in VS Code. In this case, I suppose the _auto update_ VS Code does, is doing something similar to Prettier.

Which is your scenario?

**Reply by hth2** [reply]
Hi @alefragnani,

thank you for your reply. I run `git rebase` in terminal, while the file is still opened in VS Code. So I assume that I must close the file in VS Code before running `git rebase` and then open it again, and all will be fine?

**Reply by alefragnani** [reply]
Yep, that's the idea. If my assumption are correct, the bookmarks should remain.
